cmd/substreams/run.go

```diff
@@ -50,13 +51,24 @@ func runRun(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("read manifest %q: %w", manifestPath, err)
 	}
 
-	outputModule := args[1]
+	productionMode := mustGetBool(cmd, "production-mode")
+	debugModulesOutput := mustGetStringSlice(cmd, "debug-modules-output")
+	if debugModulesOutput != nil && productionMode {
+		return fmt.Errorf("cannot set 'debug-modules-output' in 'production-mode'")
+	}
+
+	debugModulesInitialSnapshot := mustGetStringSlice(cmd, "debug-modules-initial-snapshot")
+	if debugModulesInitialSnapshot != nil && productionMode {
+		return fmt.Errorf("cannot set 'debug-modules-initial-snapshot' in 'production-mode'")
+	}
```

Comment:
I would not do this validation here, I would move this into the request validation, So that everyone else can benefit from it. That is what colin was working on. So you create the request naively and on line 111 it will fail out there.